gitui crashes while try to show branches

**Describe the bug**
just open the gitui under my repo directory, and when I try to switch to the branches view by pressing the B button on keyboard, the gitui crashes. and logs the error :

```
panic: PanicInfo { payload: Any, message: Some(assertion failed: self.is_char_boundary(new_len)), location: Location { file: "/rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b/library/alloc/src/string.rs", line: 1132, col: 13 } }
trace:
   0: __mh_execute_header
   1: __mh_execute_header
   2: __mh_execute_header
   3: __mh_execute_header
   4: __mh_execute_header
   5: __mh_execute_header
   6: __mh_execute_header
   7: __mh_execute_header
   8: __mh_execute_header
   9: __mh_execute_header
  10: __mh_execute_header
  11: __mh_execute_header
  12: __mh_execute_header
  13: __mh_execute_header
```

**To Reproduce**
Steps to reproduce the behavior:
1. open gitui under my repo directory
2. press b
3. gitui crashed, and get error 


**Expected behavior**
I don't know, maybe show the branches tree

**Screenshots**
![image](https://user-images.githubusercontent.com/25848393/112279021-64568a00-8cbe-11eb-81a7-866473ae9894.png)


**Context (please complete the following information):**
 - OS/Distro + Version: macOS 10.15.7
 - GitUI Version 0.12.0
 - Rust version: 1.50

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,6 +40,7 @@ serde = "1.0"
 anyhow = "1.0"
 unicode-width = "0.1"
 textwrap = "0.13"
+unicode-truncate = {version="0.2", git="https://github.com/Aetf/unicode-truncate.git"}
 
 [target.'cfg(all(target_family="unix",not(target_os="macos")))'.dependencies]
 which = "4.0"

--- a/src/components/branchlist.rs
+++ b/src/components/branchlist.rs
@@ -14,7 +14,6 @@ use asyncgit::{
     CWD,
 };
 use crossterm::event::Event;
-use unicode_truncate::UnicodeTruncateStr;
 use std::{
     cell::Cell,
     convert::{TryFrom, TryInto},
@@ -26,6 +25,7 @@ use tui::{
     widgets::{Block, BorderType, Borders, Clear, Paragraph},
     Frame,
 };
+use unicode_truncate::UnicodeTruncateStr;
 
 use crate::ui::Size;
 use anyhow::Result;

--- a/src/components/branchlist.rs
+++ b/src/components/branchlist.rs
@@ -14,6 +14,7 @@ use asyncgit::{
     CWD,
 };
 use crossterm::event::Event;
+use unicode_truncate::UnicodeTruncateStr;
 use std::{
     cell::Cell,
     convert::{TryFrom, TryInto},
@@ -334,7 +335,7 @@ impl BranchListComponent {
             let mut commit_message =
                 displaybranch.top_commit_message.clone();
             if commit_message.len() > commit_message_length {
-                commit_message.truncate(
+                commit_message.unicode_truncate(
                     commit_message_length
                         .saturating_sub(THREE_DOTS_LENGTH),
                 );
@@ -343,7 +344,7 @@ impl BranchListComponent {
 
             let mut branch_name = displaybranch.name.clone();
             if branch_name.len() > branch_name_length {
-                branch_name.truncate(
+                branch_name.unicode_truncate(
                     branch_name_length
                         .saturating_sub(THREE_DOTS_LENGTH),
                 );

